sklearn/metrics/classification.py

```diff
@@ -240,6 +240,8 @@ def confusion_matrix(y_true, y_pred, labels=None, sample_weight=None):
         labels = unique_labels(y_true, y_pred)
     else:
         labels = np.asarray(labels)
+        if np.all([l not in y_true for l in labels]):
```

Comment:
No, you're probably right that I presumed incorrectly. Obviously the size of `labels` is also a factor.